src/manifest.rs

```diff
@@ -289,19 +294,38 @@ impl Cargo {
 
     fn take_assets(&self, options: &Config, assets: Option<Vec<Vec<String>>>, targets: &[CargoMetadataTarget], readme: Option<&String>) -> CDResult<Vec<Asset>> {
         Ok(if let Some(assets) = assets {
-            assets.into_iter().map(|mut v| {
+            let mut all_assets = Vec::with_capacity(assets.len());
+            for mut v in assets.into_iter() {
                 let mut v = v.drain(..);
-                let mut source_file = PathBuf::from(v.next().ok_or("missing path for asset")?);
-                if source_file.starts_with("target/release") {
-                    source_file = options.path_in_build(source_file.strip_prefix("target/release").unwrap());
+                let source_path = v.next().ok_or("missing path for asset")?;
+                let target_path = PathBuf::from(v.next().ok_or("missing target for asset")?);
+                let mode = u32::from_str_radix(&v.next().ok_or("missing chmod for asset")?, 8)
+                    .map_err(|e| CargoDebError::NumParse("unable to parse chmod argument", e))?;
+                if is_glob_pattern(&source_path) {
+                    for entry in glob::glob(&source_path)? {
+                        let mut source_file = entry?.clone();
+                        if source_file.is_dir() {
+                            continue;
+                        }
+                        if source_file.starts_with("target/release") {
```

Comment:
shouldn't that apply to both globs and non-globs? 

Also target/release may not exist (when cross-compiling it's `target/arch/release`), so the path fixup probably needs to be done before the call to glob, so that the glob scans appropriate dir.